test/Google.Api.Gax.Tests/ResourceNameListTest.cs

```diff
@@ -0,0 +1,131 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Google.Api.Gax.Tests
+{
+    public class ResourceNameListTest
+    {
+        private IList<string> List(params string[] items) => new List<string>(items);
+        private ResourceNameList<UnknownResourceName> ResourceList(IList<string> list) =>
```

Comment:
Some tests need to have the underlying IList, hence constructing the ResourceNameList in two parts.